### DISCUSSION TITLE
Ezra Bible App 1.4

### DISCUSSION THREAD

**Discussion by tobias-klein** [ezra-bible-app/ezra-bible-app#575] [General]
**Ezra Bible App 1.4** features an enhanced fullscreen mode with a circular verse context menu, new side buttons for toggling the tag list and dictionary, an optimized tag list view as well as various enhancements and bugfixes.

Note-worthy features, improvements and bugfixes are:
* Fullscreen mode with hidden menu and wheel navigation as verse context menu. ([#454][i454])
* Made sidebar (tag view) and dictionary switchable by side buttons. ([#346][i346])
* Use only one toolbox window for tag list and dictionary. ([#542][i542])
* Added search scope option for module search functionality. ([#518][i518])
* Fixed bug: Repository update failed already if only one repository was offline. ([#395][i395])
* Fixed bug: Raised SWORD InstallMgr default timeout from 10s to 20s to fix issues with slow internet connections. ([#157][i157])

A detailed listing of all changes can be found in the [changelog](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/CHANGELOG.md).

Installation instructions can be found [here](https://ezrabibleapp.net/installation).

[i454]: https://github.com/ezra-bible-app/ezra-bible-app/pull/454
[i346]: https://github.com/ezra-bible-app/ezra-bible-app/issues/346
[i542]: https://github.com/ezra-bible-app/ezra-bible-app/issues/542
[i518]: https://github.com/ezra-bible-app/ezra-bible-app/issues/518
[i395]: https://github.com/ezra-bible-app/ezra-bible-app/issues/395
[i157]: https://github.com/ezra-bible-app/ezra-bible-app/issues/157

<hr /><em>This discussion was created from the release <a href='https://github.com/ezra-bible-app/ezra-bible-app/releases/tag/1.4.0'>Ezra Bible App 1.4</a>.</em>

**Comment by guiquimthe** [comment]
![Captura de tela 20-12-2021 15 45 31](https://user-images.githubusercontent.com/81565303/146817368-60080b87-fd02-42ef-91bb-7cd3f9814fa8.png)
Hey guys!

At least on Windows 10, it's not functional. It's the third time I've proceeded with the Ezra Bible installer 1.4 and after installing it remains at infinite loading.

**Reply by tobias-klein** [reply]
@guiquimthe Thank you for reporting this! I'll have a look tomorrow. Since this seems like a critical bug, I'll publish a bugfix release right away if I find the root-cause!

**Reply by guiquimthe** [reply]
@tobias-klein Thanks for the feedback Tobias! God bless.

**Reply by tobias-klein** [reply]
What failed at startup was the sqlite3 module based on a GitHub build issue. I just provided a patch release, which I have tested successfully on my Windows 10 machine. Please download it from [here](https://github.com/ezra-bible-app/ezra-bible-app/releases/download/1.4.0/ezra-bible-app_win32_setup-1.4.1.exe) and provide feedback whether it works.

## Details regarding the root-cause
It turns out that this was caused by an incompatibility between sqlite3 and node 16. Generally we are using node 14 for all builds and that was also the case until most recently with GitHub generated Windows builds (and I had tested those). However, GitHub recently [switched to node 16 on their Windows build environment](https://github.com/actions/virtual-environments/commit/92269a3a22c96240dfb589ee40adc66be12dbba5#diff-816219d3c9f30fcc1c40f88ca826c1662bf99eded781c1b419dd2dfb577ae227R21). This introduced the mentioned incompatibility with sqlite3 and therefore the build partly failed. This was now fixed by fixing the node version to 14.x on the GitHub Actions Windows environment.

**Reply by guiquimthe** [reply]
Thank you very much for resolving this issue. It looks great now and installed very fast.

**Comment by pepaulo** [comment]
I don't see the deuterocanonical books in the app. Show only protestant canon. How I can change it?

**Reply by tobias-klein** [reply]
Deuterocanonical books are currently not supported by the app. The simple reason is that there is a static book list in the menu and in other places. We would simply need to extend this list and make the menu a bit more dynamic. Would you like to open an issue for this?

**Reply by tobias-klein** [reply]
@pepaulo I added issue #612 and will try to integrate this for the upcoming release.
